shai/src/main/scala/at/forsyte/apalache/shai/TransExplorerService.scala

```diff
@@ -40,10 +68,75 @@ class TransExplorerService(connections: Ref[Map[UUID, Conn]]) extends ZioTransEx
    * @param req
    *   the request (isomorphic to the Unit)
    */
-  def openConnection(req: ConnectRequest): ZIO[ZEnv, Status, Connection] = for {
+  def openConnection(req: ConnectRequest): Result[Connection] = for {
     id <- ZIO.effectTotal(UUID.randomUUID())
     _ <- addConnection(Conn(id))
   } yield Connection(id.toString())
 
-  private def addConnection(c: Conn): UIO[Unit] = connections.update(_ + (c.id -> c))
+  /**
+   * Parses a spec into a model
+   *
+   * This method handles the [LoadModelRequest] RPC defined by `transExplorer.proto`
+   *
+   * @param req
+   *   the request to load a model, including the root module spec and any auxiliary modules
+   */
+  def loadModel(req: LoadModelRequest): Result[LoadModelResponse] = for {
+    parseResult <- parseSpec(req.spec, req.aux)
+    result <- parseResult match {
+      case Right(module) =>
+        for {
+          _ <- updateConnection(req.conn)(_.setModel(module))
+          json = jsonOfModule(module)
+        } yield LoadModelResponse.Result.Spec(json)
+      case Left(err) =>
+        ZIO.succeed(LoadModelResponse.Result.Err(err.getMessage()))
```

Comment:
This is probably my ignorance of the ZIO effect system, but why is only the `Left` branch wrapped in `ZIO.succeed`?